README


This archive contains the distribution Getopt-Modular,
version 0.12:

  Modular access to Getopt::Long

This software is copyright (c) 2014 by Darin McBride.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.008.



Changes
Revision history for Getopt-Modular

0.12    2014-01-06
        ! Missing proper check for subtypes	

0.11    2013-05-25
        ! missing 'valid' field for valid_value exceptions
        ! Valid values in help expected array ref instead of array, fixed.
        + Add optional hash-ref parameter to parseArgs to return all the
          arguments, whether set or by default or whatever.  Not recommended
          for new code, but may make using this in apps that already use
          Getopt::Long easier as a shim for conversion.

0.10    2012-10-30
        + Remove extra spaces at the end of getHelpWrap that Text:Table
          puts in, in case one line is too long for the screen, no need for
          all lines to be too long for the screen.
        + getHelpWrap also now wraps current/valid values if the default/valid
          values are too wide for the screen with its leading text.

0.09    2012-09-19
        + Add "hidden" flag for parameters that should be accepted but
          not displayed in getHelp/getHelpWrap

0.08    2012-07-24
        + Add exception types.  Theoretically good for translatability
          (but you get to handle translations and catching everything).
        + Add valid_values option
        + getHelpWrap will use Text::WrapI18N if available, otherwise use
          Text::Wrap (core).
        + Allow callbacks for "current value" and "valid value" strings in
          getHelp and getHelpWrap

0.07    June 3, 2011
        * Fix some doc typos, add more explanation on how defaults are dynamic
        * Reorder revision history: new stuff at top.
        + Added import options: -namespace, -getOpt

0.06    October 28, 2010
        ! Remove the rest of the dependency on Contextual::Return since it
          no longer seems to work on perl 5.8.8

0.05    September 1, 2010
        ! Gotta run all tests prior to shipping.  Fixed an erroneous test.

0.04    August 31, 2010
        ! Fixed some tests: kwalitee didn't eval right, help didn't skip right.
        + Added some tests to improve test coverage
        ! Removed some (not all) dependency on Contextual::Return where it
          didn't seem to want to work properly.
        - Removed dependency on Smart::Comments since they're only used for
          debugging, not runtime.

0.03    July 24, 2008
        ! Removed some boilerplate
        * Defaults for boolean now show up in English rather than
          perl-bool (undef vs 1) in the help
        + Settable boolean text
        + Added getHelpWrap for Text::Wrapped help.

0.02    July 23, 2008
        ! Fixed missing pre-req
        + Added Changes 

0.01    July 22, 2008
        + Core functionality released




Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.008.
use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Modular access to Getopt::Long",
  "dist_author" => [
    "Darin McBride <dmcbride\@cpan.org>"
  ],
  "dist_name" => "Getopt-Modular",
  "dist_version" => "0.12",
  "license" => "perl",
  "module_name" => "Getopt::Modular",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Exception::Class" => 0,
    "Getopt::Long" => 0,
    "List::MoreUtils" => 0,
    "Scalar::Util" => 0,
    "Text::Table" => 0,
    "Text::Wrap" => 0,
    "Text::WrapI18N" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Data::Dumper" => 0,
    "Test::Deep" => 0,
    "Test::Exception" => 0,
    "Test::More" => "0.88"
  }
);


my %fallback_build_requires = (
  "Data::Dumper" => 0,
  "Module::Build" => "0.3601",
  "Test::Deep" => 0,
  "Test::Exception" => 0,
  "Test::More" => "0.88"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.008.
use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Modular access to Getopt::Long",
  "AUTHOR" => "Darin McBride <dmcbride\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "CONFIGURE_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "Getopt-Modular",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Getopt::Modular",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Exception::Class" => 0,
    "Getopt::Long" => 0,
    "List::MoreUtils" => 0,
    "Scalar::Util" => 0,
    "Text::Table" => 0,
    "Text::Wrap" => 0,
    "Text::WrapI18N" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Data::Dumper" => 0,
    "Test::Deep" => 0,
    "Test::Exception" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "0.12",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Data::Dumper" => 0,
  "Exception::Class" => 0,
  "Getopt::Long" => 0,
  "List::MoreUtils" => 0,
  "Module::Build" => "0.3601",
  "Scalar::Util" => 0,
  "Test::Deep" => 0,
  "Test::Exception" => 0,
  "Test::More" => "0.88",
  "Text::Table" => 0,
  "Text::Wrap" => 0,
  "Text::WrapI18N" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



